src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs

```diff
@@ -85,6 +85,26 @@ public StringValues(string?[]? values)
             return value.GetArrayValue();
         }
 
+#if NET9_0_OR_GREATER
+        /// <summary>
+        /// Defines an implicit conversion of a given <see cref="StringValues"/> to a string span.
+        /// </summary>
+        /// <param name="value">A <see cref="StringValues"/> to implicitly convert.</param>
+#pragma warning disable CS3006 // Overloaded method 'StringValues.implicit operator ReadOnlySpan<string?>(in StringValues)' differing only in ref or out, or in array rank, is not CLS-compliant.
+        public static implicit operator ReadOnlySpan<string?>(in StringValues value)
+#pragma warning restore CS3006
+        {
+            if (value._values is string)
+            {
+                return new ReadOnlySpan<string?>(in Unsafe.As<object, string?>(ref Unsafe.AsRef(in value._values)));
```

Comment:
Isn't this extremely unsafe?

Other threads (or even the same thread) could overwrite the reference.
The rest of `StringValues` is careful to always do `Unsafe` only on locals that it already type checked.

```c#
_field = new("foo");
ReadOnlySpan<string> span = _field;
_field = new(["bar"]);

if (span[0].GetType() != typeof(string))
{
    Console.WriteLine("Oh no");
}
```

Reply:
That's a very good catch.

And I don't see a good way to achieve what the method needs to achieve and still be able to make it safe.

I think that means we need to scrap this API.

Reply:
> we need to scrap this API.

@stephentoub don't we want change to the allocation form?

Reply:
> > we need to scrap this API.
> 
> @stephentoub don't we want change to the allocation form?

I'd be ok with that, but there were several folks who voiced opposition to having a span-returning API like this that allocated.